src/App.tsx

```diff
@@ -17,18 +55,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setFilterBy={setFilterBy}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+
+              <TodoList
+                todos={filteredTodos}
+                selectedTodo={selectedTodo}
+                setSelectedTodo={setSelectedTodo}
+              />
```

Comment:
Use a ternary operator here, so you don't render empty list while loading is true